pkg/operators/cli/clioperator.go

```diff
@@ -205,7 +205,26 @@ func (o *cliOperatorInstance) PreStart(gadgetCtx operators.GadgetContext) error
 			fields, hasFields = fieldLookup[""] // fall back to default
 		}
 
-		switch o.mode {
+		mode := o.mode
+		if annotatedMode := ds.Annotations()["cli.output"]; annotatedMode != "" {
+			mode = annotatedMode
+		}
+
+		switch mode {
+		case ModeRaw:
+			before := func() {}
+			if ds.Annotations()[AnnotationClearScreenBefore] == "true" && term.IsTerminal(int(os.Stdout.Fd())) {
+				before = clearScreen
+			}
```

Comment:
If it would be done in clearScreen, it would be done on every call. However, we could make clearScreen return a function instead - depending on whether it's a terminal or not.